src/data-table.component.html

```diff
@@ -104,8 +104,12 @@
       </ng-container>
       <!-- master-detail row definition -->
       <ng-container matColumnDef="$masterDetail" sticky>
-        <th mat-header-cell *matHeaderCellDef scope="col" class="master-detail-header"></th>
-        <td mat-cell *matCellDef="let row; let idx = index" class="master-detail-cell">
+        <th mat-header-cell *matHeaderCellDef scope="col" class="master-detail-header" [class.master-detail-header--shrink]="!showDetailHeader" [class.master-detail-header--expand]="showDetailHeader">
+          <span *ngIf="showDetailHeader">
+            {{ 'MASTER_DETAIL.HEADER' | translate }}
```

Comment:
'MASTER_DETAIL.HEADER' should be empty on default, so it does not break current functionalities

Reply:
It will only appear if you set `showDetailHeader` (which is false by default) input to true.